src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,88 @@
-export const Pagination = () => {};
+import { FC, MouseEvent } from 'react';
+
+import { getNumbers } from '../../utils';
+
+type OnPageChange = (page: number) => void;
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: OnPageChange;
+}
+
+export const Pagination: FC<Props> = props => {
+  const { total = 0, perPage, currentPage, onPageChange } = props;
+  const numberOfPages = Math.ceil(total / perPage);
+
+  const handlePageChange = (event: MouseEvent) => {
+    const page = (event.target as HTMLElement).innerText;
```

Comment:
Using `innerText` to extract the page number can be unreliable if the `innerText` is not a valid number. Consider using a data attribute or another method to reliably extract the page number.